docs: add flush accordion content

laurenmrice opened the pull request:
Closes #3147 

- Add flush accordion content and images to the usage and style tabs. 

---

**New**

Usage tab:
- New content section
- New images

Style tab:
- New images
- New content in the structure table 

**Changed**

- Changed some images.

aagonzales commented:
The hover in the image looks really dark, almost the same color as the table header. Maybe double check that one?

<img width="767" alt="image" src="https://user-images.githubusercontent.com/11670886/192844270-cb1a2a49-2b9e-4fc5-8788-4840d486873b.png">

aagonzales commented on src/pages/components/accordion/style.mdx:
```diff
@@ -56,25 +72,41 @@ There is no max-height for an open panel but an accordion may
+| Icon                     | size           | 16 / 1   | –             |
+| Icon: default alignment  | padding-right  | 16 / 1   | `$spacing-05` |
+| Icon: flush alignment    | padding-right  | 0        | –             |
+| Title                    | margin-left    | 16 / 1   | `$spacing-05` |
```
Does title needs a "default" and "flush" distinction too?

laurenmrice replied:
yes!

aagonzales reviewed (commented):
- I wonder if on the usage tab in the "[Grid placement](https://carbondesignsystem-git-fork-laurenm-2d877c-carbon-design-system.vercel.app/components/accordion/usage/#placement)" section we should link down to the modifer. 

- On the style tab, I think you should split the structure table into two sections. I think it would also help to show the structure outlines of the focus/hover state as well.

- On the style tab, what do you think about putting to two color images into a tab component (like I'm doing with Fluid) to help save space since the images are so similar?

laurenmrice commented:
> On the style tab, I think you should split the structure table into two sections... 
@aagonzales Do you mean having separate sections for the alignments in the same table? Or to have one table per alignment?

aagonzales commented:
I was kinda thinking one table per alignment. That's just a gut feeling though, its not a hill I'll die on.

laurenmrice commented:
@aagonzales okay! Let me know what you think of the latest preview.

Usage: I added a reference and link to the Modifier section in the "Grid placement" paragraph.
Style: I separated out the default/flush alignments for the Structure section and I also have two images per section to show structure versus spacing specs. I also added the tabs for the Color section.

aagonzales reviewed (approved):
Yes, i think that adds some nice clarity.
